uae4all_gp2x_0.7.2a/src/od-joy.mm

```diff
@@ -39,7 +39,23 @@
 
 #import <UIKit/UIKit.h>
 #import <GameController/GameController.h>
+#import "MainEmulationViewController.h"
 
+
+
+MainEmulationViewController *mainViewController=nil;
+void set_MainView(MainEmulationViewController *m)
+{
```

Comment:
Do you really need to set the main view controller on this lower level class?  Seems like it is only used below as a nil check?